packages/fhi-designsystem/src/components/fhi-radio/fhi-radio.component.ts

```diff
@@ -0,0 +1,301 @@
+    if (changedProperties.has('checked')) {
+      this._input.checked = !!this.checked;
+      this._setFormValue();
+
+      if (this.checked) {
+        this.uncheckGroupMembers();
+      }
+    }
+
+    // update the form value when the value or name input changes and the radio is already checked
+    if (changedProperties.has('value') || changedProperties.has('name')) {
+      if (this._input.checked) {
+        this._setFormValue();
+      }
+    }
+  }
+
+  public formResetCallback(): void {
+    this.checked = typeof this.getAttribute('checked') === 'string';
+
+    this._setFormValue();
+
+    if (this.checked) {
+      this.uncheckGroupMembers();
+    }
+  }
+
+  private uncheckGroupMembers(): void {
+    const root = this._internals.form ? this._internals.form : document;
+
+    const radios = root.querySelectorAll<FhiRadio>(
+      `${FhiRadioSelector}[name="${this.name}"]`,
+    );
+
+    radios.forEach(radio => {
+      if (radio === this) {
+        return;
+      }
+
+      if (radio.isFormElement && root === document) {
+        return;
+      }
+
+      radio.checked = false;
+    });
+  }
+
+  private _setFormValue(): void {
+    if (!this.isFormElement) {
+      return;
+    }
+
+    this._internals.setFormValue(this.checked ? this.value : null);
+  }
+
+  private _handleChange(event: Event): void {
+    this.checked = (event.target as HTMLInputElement).checked;
+
+    this.uncheckGroupMembers();
+
+    this._setFormValue();
+
+    event.stopPropagation();
+
+    this.dispatchEvent(new Event('change', { bubbles: true }));
+    this.dispatchEvent(new Event('input', { bubbles: true, composed: true }));
+  }
+
+  private _handleInput(event: Event): void {
+    event.stopPropagation();
+  }
+
+  render() {
+    return html`
+      <div id="radio-container">
+        <input
+          type="radio"
+          id="input-element"
+          name="${this.name}"
+          value="${this.value}"
+          ?checked=${this.checked}
+          ?disabled=${this.disabled}
+          @change=${this._handleChange}
+          @input=${this._handleInput}
+        />
+        <svg
+          class="radio-dot"
+          width="18"
+          height="18"
+          viewBox="0 0 18 18"
+          xmlns="http://www.w3.org/2000/svg"
+        >
+          <circle r="6" cx="9" cy="9" />
+        </svg>
+      </div>
+      ${this.label && html`<label for="input-element">${this.label}</label>`}
+    `;
+  }
+
+  static styles = css`
+    :host {
+      --motion-radio-transition: var(--fhi-motion-ease-default)
+        var(--fhi-motion-duration-quick);
+
+      --color-radio-label: var(--fhi-color-neutral-text-default);
+      --color-radio-border: var(--fhi-color-neutral-border-default);
+      --color-radio-background: var(--fhi-color-neutral-background-default);
+
+      --color-radio-border-hover: var(--fhi-color-accent-border-strong);
+      --color-radio-background-hover: var(--fhi-color-accent-background-subtle);
```

Comment:
```suggestion
      --color-radio-background-hover: var(--fhi-color-accent-background-subtle);
      --color-radio-background-active: var(--fhi-color-accent-surface-default);
```

Må også ha tilsvarende for error, og settes på :active-state.

Brukte active siden pseudoklassen heter det, og fordi dette samsvarer med variabler i Checkbox.